### PR TITLE
Adding a Highway layer

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: allennlp/modules/highway.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self._layers = torch.nn.ModuleList([torch.nn.Linear(input_dim, input_dim * 3)
+                                            for _ in range(num_layers)])
+        self._activation = activation
+
+    @overrides
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:  # pylint: disable=arguments-differ
+        current_input = inputs
+        for layer in self._layers:
+            projected_input = layer(current_input)
+            linear_part = projected_input[:, 0 * self._input_dim : 1 * self._input_dim]
```
Comment (DeNeutoy) [inline allennlp/modules/highway.py]: I don't think having the additional 1's and 0's here makes it clearer - it's not particularly complicated slicing anyway.

File: tests/modules/highway_test.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+from allennlp.modules import Highway
+from allennlp.testing import AllenNlpTestCase
+
+
+class TestHighway(AllenNlpTestCase):
+    def test_forward_works_on_simple_input(self):
+        highway = Highway(2, 2)
+        # pylint: disable=protected-access
+        highway._layers[0].weight.data.fill_(1)
```
Comment (DeNeutoy) [inline tests/modules/highway_test.py]: Did you not use allennlp initialisers here for some particular reason? Maybe we need to rethink them.

File: allennlp/modules/highway.py (hunk 1 above)
Comment (matt-gardner) [inline allennlp/modules/highway.py]: I was following what you did in your augmented LSTM.  I think this is marginally more clear than dropping them, but I don't have a strong opinion here.

File: tests/modules/highway_test.py (hunk 2 above)
Comment (matt-gardner) [inline tests/modules/highway_test.py]: For a simple test, this is a lot easier than building up an `InitializerApplicator`, which would need to have several cases.

How were you imagining using the applicator?  Seems like the right place to have initialization is inside model code.

File: allennlp/modules/highway.py (hunk 1 above)
Comment (DeNeutoy) [inline allennlp/modules/highway.py]: Ah, oh yeah. My bad.

**Pull request merged**